crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -154,24 +179,36 @@ impl<S: States> Deref for State<S> {
 ///     next_game_state.set(GameState::InGame);
 /// }
 /// ```
-#[derive(Resource, Debug)]
+#[derive(Resource, Debug, Default)]
 #[cfg_attr(
     feature = "bevy_reflect",
     derive(bevy_reflect::Reflect),
-    reflect(Resource, Default)
+    reflect(Resource)
 )]
-pub struct NextState<S: States>(pub Option<S>);
-
-impl<S: States> Default for NextState<S> {
-    fn default() -> Self {
-        Self(None)
-    }
+pub enum NextState<S: FreelyMutableState> {
+    /// No transition has been planned
+    #[default]
+    Unchanged,
+    /// There is a transition planned for state `S`
+    Set(S),
+    /// There is a planned removal of the state `S`
+    Remove,
```

Comment:
I thought we agreed to remove this from this PR

Reply:
Misunderstood/missed on my part, as discussed on discord. removed now, and will open a separate PR later.